torch/_dynamo/backends/cppjit.py

```diff
@@ -0,0 +1,964 @@
+
+
+def type_aligned_arguments(
+    aligned_arguments: Sequence[AlignedArg], sig: Signature
+) -> List[CTypedAlignedArg]:
+    """Associates the corresponding parameter CType to each argument.
+
+    This function assumes each argument (FunctionSchema parameter) has
+    only one corresponding Binding, which is the source of the CType.
+    """
+    param_to_bindings = groupby(lambda b: b.argument, sig.arguments())
+
+    for param, bindings in param_to_bindings.items():
+        assert (
+            len(bindings) == 1
+        ), f"unsupported multi-binding parameter {param} with bindings: {bindings}"
+
+    return [
+        CTypedAlignedArg(a, ctype=param_to_bindings[a.param][0].nctype.type)
+        for a in aligned_arguments
+    ]
+
+
+def py_to_cppstr(thing: Any, ty: CType) -> str:
+    """Parses the a Python value of a given type into a C++ string."""
+    if isinstance(thing, torch.fx.Node):
+        return thing.name
+
+    if (
+            ty == BaseCType(boolT)
+            or (isinstance(thing, bool) and ty == BaseCType(scalarT))
+    ):
+        return str(thing).lower()
+
+    if ty == BaseCType(stringT):
+        return f'"{thing}"'
+
+    if ty == BaseCType(deviceT):
+        return f"""at::Device("{thing}")"""
+
+    if isinstance(ty, BaseCType) and ty.type in (layoutT, memoryFormatT, scalarTypeT):
+        return ENUM_CPPSTR_DISPATCH[ty.type][thing]
+
+    if isinstance(ty, (ArrayCType, ListCType)):
+        assert isinstance(thing, list)
+        thing_str = ", ".join(py_to_cppstr(x, ty.elem) for x in thing)
+        return f"{ty.cpp_type()}({{{thing_str}}})"
+
+    if (
+        isinstance(ty, BaseCType)
+        and ty.type
+        in (
+            intArrayRefT,
+            iTensorListRefT,
+            iOptTensorListRefT,
+            symIntArrayRefT,
+            tensorListT,
+        )
+    ) or (isinstance(ty, ArrayRefCType) and isinstance(ty.elem, BaseCType)):
+        assert isinstance(thing, list)
+        cpptype = ELEM_TYPE_FOR[ty.type] if isinstance(ty, BaseCType) else ty.elem
+        thing_str = ", ".join(py_to_cppstr(x, cpptype) for x in thing)
+        return f"at::ArrayRef<{cpptype.cpp_type()}>({{{thing_str}}})"
+
+    if (
+        isinstance(ty, BaseCType)
+        and ty.type
+        in (
+            optionalIntArrayRefT,
+            optionalScalarRefT,
+            optionalSymIntArrayRefT,
+            optionalTensorRefT,
+        )
+    ) or isinstance(ty, OptionalCType):
+        if thing is None:
+            return "c10::nullopt"
+
+        elem_ty = ELEM_TYPE_FOR[ty.type] if isinstance(ty, BaseCType) else ty.elem
+        return py_to_cppstr(thing, elem_ty)
+
+    if isinstance(ty, BaseCType):
+        if ty == BaseCType(tensorT):
+            assert isinstance(thing, (bool, int, float)), f"unsupported scalar value: {thing} ({type(thing)})"
+            return f"THPVariable_Unpack(THPVariable_Wrap(at::native::wrapped_scalar_tensor({py_to_cppstr(thing, PYTYPE_TO_CTYPE[type(thing)])})))"
```

Comment:
What's this pattern?
```suggestion
            return f"at::native::wrapped_scalar_tensor({py_to_cppstr(thing, PYTYPE_TO_CTYPE[type(thing)])})"
```